src/pages/write-smart-contracts/setting-up-local-environment.md

```diff
@@ -0,0 +1,201 @@
+---
+title: Setting up local environment
+description: Learn how to set up your local smart contract development environment
+duration: 30 minutes
+experience: intermediate
+---
+
+## Introduction
+
+As soon as you get to work on your smart contracts and testing them end to end within your application, you need a local development environment that supports all steps and helps you achieve faster iteration cycles. This is critical because transaction times on the testnet take a few minutes and would slow you down. The solution is to run a local version of the testnet, often called a "mocknet".
+
+By the end of this tutorial, you will:
+
+- Have all required tooling and services up and running locally
+- Be able to deploy and test smart contracts inside your app
+
+## Prerequisites
+
+### Prior tutorials
+
+You need to be familiar with Clarity and the Stacks authentication (using Connect). Please finish all previous tutorials, especially the [Counter tutorial](/smart-contracts/counter-tutorial) and the [Building a todo app tutorial](/authentication/building-todo-app).
+
+### Environment
+
+You should have [NodeJS](https://nodejs.org/en/download/) and [Docker](https://docs.docker.com/get-docker/) installed on your workstation.
+
+## Overview
+
+For a complete local mocknet setup, you need the following services:
+
+- [Stacks Node](https://github.com/blockstack/stacks-blockchain)
+- [Stacks Blockchain API](https://github.com/blockstack/stacks-blockchain-api)
+- [Stacks Explorer](https://github.com/blockstack/explorer)
+
+## Step 1: Spin up mocknet
+
+The easiest way to spin up the Node, Blockchain API, and the Explorer locally is with Docker:
+
+```bash
+git clone https://github.com/blockstack/stacks-local-dev
+cd stacks-local-dev/
+docker-compose up -d
+```
+
+-> Make sure your Docker service is up and running
+
+The initial start will take a few minutes. As soon as it is complete, you should be able to open up [http://localhost:3999/](http://localhost:3999/) and see:
+
+```js
+{"status":"ready"}
+```
+
+~> The Stacks Node is running on port `20443`, the Stacks Blockchain API on `3999` and the Stacks Explorer is on port `3000`.
```

Comment:
When I pulled the latest repo, the explorer is disabled by default. we should note that.